src/Simulation/Simulators/SparseSimulator/Native/SparseSimulator.h

```diff
@@ -0,0 +1,846 @@
+#pragma once
+#include <string>
+#include "types.h"
+#include "gates.h"
+#include <random>
+#include <cmath>
+#include <functional>
+#include "quantum_state.hpp"
+#include "basic_quantum_state.hpp"
+#include <algorithm>
+#include <iomanip>
+#include <iostream>
+#include <list>
+
+#include <set>
+
+
+#define MAX_QUBITS 1024
+#define MIN_QUBITS 64
```

Comment:
FYI: [Item 2: Prefer `const`s, `enum`s, and `inline`s to `#defines`](http://wavelino.coffeecup.com/pdf/Effective%20Cshapshap.pdf)

Reply:
Yes, defines should be avoided. Replaced with constexpr here: https://github.com/sam-jaques/qsharp-runtime/pull/1